### DISCUSSION TITLE
Integration of auto-scaling for mimir microservices

### DISCUSSION THREAD

**Discussion by qdupuy** [grafana/mimir#2672] [Help and support]
Hello, 🖖 

In the context of a highly available platform, is it possible to implement HPA and VPA for mimir microservices?

Indeed, when we receive a quantity of additional flow, it would be great that the services increase on its number of replicas

**Comment by pracucci** [comment]
Hi 🖖 !

> is it possible to implement HPA and VPA for mimir microservices?

Yes, and HPA is something we're actively working on. Mimir jsonnet supports autoscaling for `querier` and `ruler-querier` ([see code](https://github.com/grafana/mimir/blob/main/operations/mimir/autoscaling.libsonnet)), based on [KEDA](https://keda.sh/). Mimir Helm chart will follow.

**Reply by shybbko** [reply]
@pracucci any update / roadmap on Helm-native HPA? I've just migrated from Cortex and it seems that HPA is supported for nginx pods only, while Cortex did run HPA fine for multiple deployments. In my scenario I deploy things declared in Helm via Terraform, so switching to Jsonnet and Tanka is suboptimal, having HPA available in a Helm-native code would be perfect.

**Reply by dimitarvdimitrov** [reply]
hey @shybbko, there is an open issue for adding autoscaling to `mimir-distributed` - https://github.com/grafana/mimir/issues/3430 

It also has details on how to do this yourself using the generated manifests from the jsonnet libs. At present we don't have plans to work on this. This is not set in stone and we're also open to reviewing contributions for autoscaling in Helm.
